### ISSUE TITLE
Neo4j server fails to execute statement with "java.lang.IllegalStateException: NoLock[Pipeline[3]] is already locked"

### ISSUE THREAD

**Issue by rjbaucells** [neo4j/neo4j#12700]
- Neo4j version: 4.2.4-enterprise
- Operating system: Linux (docker)
- API/Driver: Neo4J Browser

Steps:

* Make sure you have a working docker installation

* Download the [database](https://dcr3gddjqn6ms.cloudfront.net/neo4j-local.zip) (~250MB)

* Unzip package

```bash
unzip neo4j-local.zip
```

* Start neo4j database server

```bash
cd neo4j-local
./run.sh
```

* Open [neo4j browser](http://localhost:7474/browser/), username: `neo4j`, password: `neo4jbug`

* Execute the following CYPHER statements:

```cypher
:use production
```

```cypher
MATCH (_3:`JournalEntryLine` {`$tenant`: '100'}) 
MATCH (:`__UNIQUE_VERTEX_VALUE__` {`$tenant`: '100', `$uniqueValue`: _3.`$_HAS_CATEGORY_VALUE`})<-[:`__HAS_UNIQUE_VALUE__`]-(_1:`Category` {`$tenant`: '100'}) WHERE (_1.`name` = 'EE4D5BFA-A708-40A3-9385-4EC3DB779EB0') 
MATCH (:`__UNIQUE_VERTEX_VALUE__` {`$tenant`: '100', `$uniqueValue`: _3.`$_HAS_GL_ACCOUNT_VALUE`})<-[:`__HAS_UNIQUE_VALUE__`]-(_2:`GLAccount` {`$tenant`: '100'}) WHERE (_2.`number` = '30001')
RETURN ID(_3)
```

* Expected behavior

Execution without errors

- Actual behavior

Database server fails to execute statement, `debug.log` shows the following exception:

```
2021-03-31 15:32:30.191+0000 ERROR [o.n.f.t.ErrorReporter] Unexpected error [Neo.DatabaseError.General.UnknownError]: Unhandled error.
2021-03-31 15:32:30.193+0000 ERROR [o.n.f.t.ErrorReporter] Unexpected error [Neo.DatabaseError.General.UnknownError]: Unhandled error.
java.lang.IllegalStateException: NoLock[Pipeline[3]] is already locked
	at org.neo4j.cypher.internal.runtime.pipelined.state.NoLock.tryLock(Lock.scala:40) ~[neo4j-cypher-pipelined-runtime-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.runtime.pipelined.state.TheExecutionState.tryLock(TheExecutionState.scala:235) ~[neo4j-cypher-pipelined-runtime-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.runtime.pipelined.state.TheExecutionState.$anonfun$closeOutstandingWork$1(TheExecutionState.scala:316) ~[neo4j-cypher-pipelined-runtime-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.runtime.pipelined.state.TheExecutionState.$anonfun$closeOutstandingWork$1$adapted(TheExecutionState.scala:313) ~[neo4j-cypher-pipelined-runtime-4.2.4.jar:4.2.4]
	at scala.collection.Iterator.foreach(Iterator.scala:941) ~[scala-library-2.12.10.jar:?]
	at scala.collection.Iterator.foreach$(Iterator.scala:941) ~[scala-library-2.12.10.jar:?]
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1429) ~[scala-library-2.12.10.jar:?]
	at scala.collection.IterableLike.foreach(IterableLike.scala:74) ~[scala-library-2.12.10.jar:?]
	at scala.collection.IterableLike.foreach$(IterableLike.scala:73) ~[scala-library-2.12.10.jar:?]
	at scala.collection.AbstractIterable.foreach(Iterable.scala:56) ~[scala-library-2.12.10.jar:?]
	at org.neo4j.cypher.internal.runtime.pipelined.state.TheExecutionState.closeOutstandingWork(TheExecutionState.scala:313) ~[neo4j-cypher-pipelined-runtime-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.runtime.pipelined.state.TheExecutionState.cancelQuery(TheExecutionState.scala:282) ~[neo4j-cypher-pipelined-runtime-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.runtime.pipelined.execution.CallingThreadExecutingQuery.cleanUpOnCancel(CallingThreadExecutingQuery.scala:66) ~[neo4j-cypher-pipelined-runtime-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.runtime.pipelined.execution.CallingThreadExecutingQuery.cancel(CallingThreadExecutingQuery.scala:58) ~[neo4j-cypher-pipelined-runtime-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.PipelinedRuntimeResult.cancel(PipelinedRuntime.scala:453) ~[neo4j-enterprise-cypher-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.result.StandardInternalExecutionResult.close(StandardInternalExecutionResult.scala:84) ~[neo4j-cypher-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.result.ClosingExecutionResult.close(ClosingExecutionResult.scala:79) ~[neo4j-cypher-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.result.InternalExecutionResult.close(InternalExecutionResult.scala:64) ~[neo4j-cypher-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.result.InternalExecutionResult.close$(InternalExecutionResult.scala:64) ~[neo4j-cypher-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.result.ClosingExecutionResult.close(ClosingExecutionResult.scala:48) ~[neo4j-cypher-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.result.ClosingExecutionResult.await(ClosingExecutionResult.scala:170) ~[neo4j-cypher-4.2.4.jar:4.2.4]
	at org.neo4j.fabric.stream.QuerySubject$BasicQuerySubject$1.doRequest(QuerySubject.java:191) ~[neo4j-fabric-4.2.4.jar:4.2.4]
	at org.neo4j.fabric.stream.QuerySubject$BasicQuerySubject$1.request(QuerySubject.java:167) ~[neo4j-fabric-4.2.4.jar:4.2.4]
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130) ~[reactor-core-3.3.9.RELEASE.jar:3.3.9.RELEASE]
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130) ~[reactor-core-3.3.9.RELEASE.jar:3.3.9.RELEASE]
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130) ~[reactor-core-3.3.9.RELEASE.jar:3.3.9.RELEASE]
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130) ~[reactor-core-3.3.9.RELEASE.jar:3.3.9.RELEASE]
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130) ~[reactor-core-3.3.9.RELEASE.jar:3.3.9.RELEASE]
	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.request(Operators.java:2116) ~[reactor-core-3.3.9.RELEASE.jar:3.3.9.RELEASE]
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130) ~[reactor-core-3.3.9.RELEASE.jar:3.3.9.RELEASE]
	at reactor.core.publisher.StrictSubscriber.request(StrictSubscriber.java:138) ~[reactor-core-3.3.9.RELEASE.jar:3.3.9.RELEASE]
	at org.neo4j.fabric.stream.Rx2SyncStream$RecordSubscriber.request(Rx2SyncStream.java:129) ~[neo4j-fabric-4.2.4.jar:4.2.4]
	at org.neo4j.fabric.stream.Rx2SyncStream.maybeRequest(Rx2SyncStream.java:91) ~[neo4j-fabric-4.2.4.jar:4.2.4]
	at org.neo4j.fabric.stream.Rx2SyncStream.readRecord(Rx2SyncStream.java:50) ~[neo4j-fabric-4.2.4.jar:4.2.4]
	at org.neo4j.fabric.bolt.BoltQueryExecutionImpl$QueryExecutionImpl.request(BoltQueryExecutionImpl.java:179) ~[neo4j-fabric-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.AbstractCypherAdapterStream.handleRecords(AbstractCypherAdapterStream.java:104) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.v3.messaging.ResultHandler.onPullRecords(ResultHandler.java:41) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.v4.messaging.PullResultConsumer.consume(PullResultConsumer.java:42) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.statemachine.impl.TransactionStateMachine$State.consumeResult(TransactionStateMachine.java:511) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.statemachine.impl.TransactionStateMachine$State$2.streamResult(TransactionStateMachine.java:355) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.statemachine.impl.TransactionStateMachine.streamResult(TransactionStateMachine.java:94) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.v4.runtime.InTransactionState.processStreamResultMessage(InTransactionState.java:73) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.v4.runtime.AbstractStreamingState.processUnsafe(AbstractStreamingState.java:49) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.v4.runtime.InTransactionState.processUnsafe(InTransactionState.java:60) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.v3.runtime.FailSafeBoltStateMachineState.process(FailSafeBoltStateMachineState.java:48) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.statemachine.impl.AbstractBoltStateMachine.nextState(AbstractBoltStateMachine.java:143) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.statemachine.impl.AbstractBoltStateMachine.process(AbstractBoltStateMachine.java:91) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.messaging.BoltRequestMessageReader.lambda$doRead$1(BoltRequestMessageReader.java:90) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.DefaultBoltConnection.lambda$enqueue$0(DefaultBoltConnection.java:147) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatchInternal(DefaultBoltConnection.java:236) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatch(DefaultBoltConnection.java:171) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatch(DefaultBoltConnection.java:161) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.scheduling.ExecutorBoltScheduler.executeBatch(ExecutorBoltScheduler.java:246) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.scheduling.ExecutorBoltScheduler.lambda$scheduleBatchOrHandleError$3(ExecutorBoltScheduler.java:229) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-all-4.1.55.Final.jar:4.1.55.Final]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2021-03-31 15:32:30.205+0000 ERROR [o.n.b.r.s.i.ErrorReporter] Client triggered an unexpected error [Neo.DatabaseError.Statement.ExecutionFailed]: null, reference e11ab44e-c1d3-463f-af58-7c6ae1dc15d6.
2021-03-31 15:32:30.206+0000 ERROR [o.n.b.r.s.i.ErrorReporter] Client triggered an unexpected error [Neo.DatabaseError.Statement.ExecutionFailed]: null, reference e11ab44e-c1d3-463f-af58-7c6ae1dc15d6.
org.neo4j.fabric.executor.FabricException: null
	at org.neo4j.fabric.executor.Exceptions.transform(Exceptions.java:51) ~[neo4j-fabric-4.2.4.jar:4.2.4]
	at org.neo4j.fabric.stream.Rx2SyncStream.readRecord(Rx2SyncStream.java:69) ~[neo4j-fabric-4.2.4.jar:4.2.4]
	at org.neo4j.fabric.bolt.BoltQueryExecutionImpl$QueryExecutionImpl.request(BoltQueryExecutionImpl.java:179) ~[neo4j-fabric-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.AbstractCypherAdapterStream.handleRecords(AbstractCypherAdapterStream.java:104) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.v3.messaging.ResultHandler.onPullRecords(ResultHandler.java:41) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.v4.messaging.PullResultConsumer.consume(PullResultConsumer.java:42) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.statemachine.impl.TransactionStateMachine$State.consumeResult(TransactionStateMachine.java:511) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.statemachine.impl.TransactionStateMachine$State$2.streamResult(TransactionStateMachine.java:355) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.statemachine.impl.TransactionStateMachine.streamResult(TransactionStateMachine.java:94) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.v4.runtime.InTransactionState.processStreamResultMessage(InTransactionState.java:73) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.v4.runtime.AbstractStreamingState.processUnsafe(AbstractStreamingState.java:49) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.v4.runtime.InTransactionState.processUnsafe(InTransactionState.java:60) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.v3.runtime.FailSafeBoltStateMachineState.process(FailSafeBoltStateMachineState.java:48) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.statemachine.impl.AbstractBoltStateMachine.nextState(AbstractBoltStateMachine.java:143) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.statemachine.impl.AbstractBoltStateMachine.process(AbstractBoltStateMachine.java:91) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.messaging.BoltRequestMessageReader.lambda$doRead$1(BoltRequestMessageReader.java:90) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.DefaultBoltConnection.lambda$enqueue$0(DefaultBoltConnection.java:147) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatchInternal(DefaultBoltConnection.java:236) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatch(DefaultBoltConnection.java:171) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatch(DefaultBoltConnection.java:161) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.scheduling.ExecutorBoltScheduler.executeBatch(ExecutorBoltScheduler.java:246) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at org.neo4j.bolt.runtime.scheduling.ExecutorBoltScheduler.lambda$scheduleBatchOrHandleError$3(ExecutorBoltScheduler.java:229) ~[neo4j-bolt-4.2.4.jar:4.2.4]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-all-4.1.55.Final.jar:4.1.55.Final]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.NullPointerException
	at org.neo4j.kernel.impl.index.schema.GenericKey.compareValueToInternal(GenericKey.java:304) ~[neo4j-kernel-4.2.4.jar:4.2.4]
	at org.neo4j.kernel.impl.index.schema.CompositeGenericKey.compareValueToInternal(CompositeGenericKey.java:90) ~[neo4j-kernel-4.2.4.jar:4.2.4]
	at org.neo4j.kernel.impl.index.schema.GenericKey.compareValueTo(GenericKey.java:282) ~[neo4j-kernel-4.2.4.jar:4.2.4]
	at org.neo4j.kernel.impl.index.schema.GenericKey.compareValueTo(GenericKey.java:54) ~[neo4j-kernel-4.2.4.jar:4.2.4]
	at org.neo4j.kernel.impl.index.schema.IndexLayout.compareValue(IndexLayout.java:79) ~[neo4j-kernel-4.2.4.jar:4.2.4]
	at org.neo4j.kernel.impl.index.schema.IndexLayout.compare(IndexLayout.java:65) ~[neo4j-kernel-4.2.4.jar:4.2.4]
	at org.neo4j.kernel.impl.index.schema.NativeIndexReader.isEmptyRange(NativeIndexReader.java:190) ~[neo4j-kernel-4.2.4.jar:4.2.4]
	at org.neo4j.kernel.impl.index.schema.NativeIndexReader.startSeekForInitializedRange(NativeIndexReader.java:154) ~[neo4j-kernel-4.2.4.jar:4.2.4]
	at org.neo4j.kernel.impl.index.schema.NativeIndexReader.query(NativeIndexReader.java:132) ~[neo4j-kernel-4.2.4.jar:4.2.4]
	at org.neo4j.kernel.impl.index.schema.GenericNativeIndexReader.query(GenericNativeIndexReader.java:119) ~[neo4j-kernel-4.2.4.jar:4.2.4]
	at org.neo4j.kernel.impl.newapi.Read.nodeIndexSeek(Read.java:113) ~[neo4j-kernel-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.runtime.pipelined.operators.ManyQueriesNodeIndexSeekTaskTemplate$.compositeCreateCursors(NodeIndexSeekOperator.scala:665) ~[neo4j-cypher-pipelined-runtime-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.runtime.pipelined.operators.ManyQueriesNodeIndexSeekTaskTemplate.compositeCreateCursors(NodeIndexSeekOperator.scala) ~[neo4j-cypher-pipelined-runtime-4.2.4.jar:4.2.4]
	at org.neo4j.codegen.OperatorTaskPipeline3_27.compiledOperate(Unknown Source) ~[?:?]
	at org.neo4j.cypher.internal.runtime.pipelined.operators.CompiledTask.operateWithProfile(OperatorCodeGenBaseTemplates.scala:342) ~[neo4j-cypher-pipelined-runtime-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.runtime.pipelined.PipelineTask.executeOperators(PipelineTask.scala:62) ~[neo4j-cypher-pipelined-runtime-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.runtime.pipelined.PipelineTask.executeWorkUnit(PipelineTask.scala:52) ~[neo4j-cypher-pipelined-runtime-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.runtime.pipelined.Worker.executeTask(Worker.scala:135) ~[neo4j-cypher-pipelined-runtime-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.runtime.pipelined.Worker.workOnQuery(Worker.scala:91) ~[neo4j-cypher-pipelined-runtime-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.runtime.pipelined.execution.CallingThreadExecutingQuery.request(CallingThreadExecutingQuery.scala:39) ~[neo4j-cypher-pipelined-runtime-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.PipelinedRuntimeResult.request(PipelinedRuntime.scala:448) ~[neo4j-enterprise-cypher-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.result.StandardInternalExecutionResult.request(StandardInternalExecutionResult.scala:88) ~[neo4j-cypher-4.2.4.jar:4.2.4]
	at org.neo4j.cypher.internal.result.ClosingExecutionResult.request(ClosingExecutionResult.scala:145) ~[neo4j-cypher-4.2.4.jar:4.2.4]
	at org.neo4j.fabric.stream.QuerySubject$BasicQuerySubject$1.doRequest(QuerySubject.java:184) ~[neo4j-fabric-4.2.4.jar:4.2.4]
	at org.neo4j.fabric.stream.QuerySubject$BasicQuerySubject$1.request(QuerySubject.java:167) ~[neo4j-fabric-4.2.4.jar:4.2.4]
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130) ~[reactor-core-3.3.9.RELEASE.jar:3.3.9.RELEASE]
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130) ~[reactor-core-3.3.9.RELEASE.jar:3.3.9.RELEASE]
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130) ~[reactor-core-3.3.9.RELEASE.jar:3.3.9.RELEASE]
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130) ~[reactor-core-3.3.9.RELEASE.jar:3.3.9.RELEASE]
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130) ~[reactor-core-3.3.9.RELEASE.jar:3.3.9.RELEASE]
	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.request(Operators.java:2116) ~[reactor-core-3.3.9.RELEASE.jar:3.3.9.RELEASE]
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130) ~[reactor-core-3.3.9.RELEASE.jar:3.3.9.RELEASE]
	at reactor.core.publisher.StrictSubscriber.request(StrictSubscriber.java:138) ~[reactor-core-3.3.9.RELEASE.jar:3.3.9.RELEASE]
	at org.neo4j.fabric.stream.Rx2SyncStream$RecordSubscriber.request(Rx2SyncStream.java:129) ~[neo4j-fabric-4.2.4.jar:4.2.4]
	at org.neo4j.fabric.stream.Rx2SyncStream.maybeRequest(Rx2SyncStream.java:91) ~[neo4j-fabric-4.2.4.jar:4.2.4]
	at org.neo4j.fabric.stream.Rx2SyncStream.readRecord(Rx2SyncStream.java:50) ~[neo4j-fabric-4.2.4.jar:4.2.4]
	... 25 more
```

**Comment by mnd999** [comment]
Thanks for the report, it looks like a bug s we will investigate and let you know when we have a fix. In the interim, it looks like prefixing your query with `CYPHER runtime=slotted` will work as a workaround.

**Comment by rjbaucells** [comment]
Any progress on this? 

I cannot use the workaround since my CYPHER queries are dynamically generated and I do not have control over it.

**Comment by henriknyman** [comment]
@rjbaucells A fix has already been merged some time ago and will be included in the upcoming 4.2.6 release. Unfortunately I cannot give you a release date yet, but you could reach out to Neo4j Support.

**Comment by khelkun** [comment]
After updating neo4j Desktop to 1.4.9 all my local databases cannot be started.

The Upgrade tab of each database says:

> Unsupported version
> Neo4j 4.2.3 is unfortunately not supported by Desktop since 1.3.4. To use it with Desktop again you need to manually upgrade, please follow the upgrade instructions.

But even after trying [upgrade instructions.](https://neo4j.com/docs/upgrade-migration-guide/current/migration/migrate-to-4.0/migrate-single-instance/#upgrade-single-instance).  The issue and the errors reported in the log files remain the same.

Desktop log.log reports:
```
Failed to get DBMS path through Relate Error: Relate DBMS with tag neo4j-desktop-dbms-database-0ad7a718-acc0-4034-84ec-a2f241a94b82 not found
```

All Databases neo4j.log reports:
```
ERROR Unexpected error [Neo.DatabaseError.General.UnknownError]: Unhandled error.
```

All Databases debug.log reports:
```
2021-11-09 16:31:06.349+0000 ERROR [o.n.f.t.ErrorReporter] Unexpected error [Neo.DatabaseError.General.UnknownError]: Unhandled error.
java.lang.IllegalStateException: NoLock[Pipeline[4]] is already locked
	at org.neo4j.cypher.internal.runtime.pipelined.state.NoLock.tryLock(Lock.scala:40) ~[neo4j-cypher-pipelined-runtime-4.2.3.jar:4.2.3]
	at org.neo4j.cypher.internal.runtime.pipelined.state.TheExecutionState.tryLock(TheExecutionState.scala:235) ~[neo4j-cypher-pipelined-runtime-4.2.3.jar:4.2.3]
	at org.neo4j.cypher.internal.runtime.pipelined.state.TheExecutionState.$anonfun$closeOutstandingWork$1(TheExecutionState.scala:316) ~[neo4j-cypher-pipelined-runtime-4.2.3.jar:4.2.3]
	at org.neo4j.cypher.internal.runtime.pipelined.state.TheExecutionState.$anonfun$closeOutstandingWork$1$adapted(TheExecutionState.scala:313) ~[neo4j-cypher-pipelined-runtime-4.2.3.jar:4.2.3]
...
```

I'm so sad
